.ebextensions/web.config

```diff
@@ -0,0 +1,12 @@
+option_settings:
+  - namespace: aws:elb:policies
+    option_name: ConnectionSettingIdleTimeout
+    value: 600
+
+commands:
+  set_web_tier_timeout:
+    command: |
+      if ! /opt/elasticbeanstalk/bin/get-config meta -k sqsdconfig 2>/dev/null; then
```

Comment:
I'm confused about this check. If a configuration for the Amazon simple queue service does not exist, then write a value into the Nginx configuration?

Also, not being familiar with Elastic Beanstalk, should I be concerned about idempotency? In other words, should there be a mechanism to prevent this same line from being written to the `web.conf` file twice?

Reply:
So far it's the most concise way I've found to tell if you're in a web or worker tier. Only the workers get that sqs setting so if it's not there you're in a web tier env

Reply:
Could you comment the code to explain that? I did searches on Google and GitHub and found very few references to that key, let alone why one would use it in that way. Thanks.

Also, is idempotency a potential issue? Could this script run twice on the same VM?

Reply:
I use it in https://github.com/Safecast/safecastapi/blob/master/.ebextensions/workers.config#L34 too so now you've got me wondering if I should look for a better way :)

And the command is idempotent (overwrites the file in question). It runs once per deploy which is sort of what you want since you might decide to change the ELB timeout which you then want nginx to pick up on.

Reply:
Went with some comments :)